src/os/bluestore/BlueStore.cc

```diff
@@ -14538,6 +14584,13 @@ void BlueStore::_do_truncate(
 
   o->onode.size = offset;
 
+  if (bdev->is_smr()) {
```

Comment:
shouldn't we put this block inside if(offset < o->onode.size) one?

Reply:
I think it should be outside so that if we get extend requests through truncate the assertion triggers.  At least for now, I haven't thought out how things should work if someone tries to extend an object through truncate.